vizro-core/src/vizro/models/_page.py

```diff
@@ -103,10 +94,34 @@ def __init__(self, **data):
                 f"as the page title. If you have multiple pages with the same title then you must assign a unique id."
             ) from exc
 
+    def _get_page_actions_chains(self) -> List[ActionsChain]:
+        """Gets all ActionsChains present on the page."""
+        page_actions_chains = []
+
+        for model_id in model_manager._get_model_children(model_id=ModelID(str(self.id))):
+            model = model_manager[model_id]
+            if hasattr(model, "actions"):
+                page_actions_chains.extend(model.actions)
+
+        for control in self.controls:
+            if hasattr(control, "selector") and control.selector:
+                page_actions_chains.extend(control.selector.actions)
+
+        return page_actions_chains
+
+    def _get_page_model_ids_with_figure(self) -> List[ModelID]:
+        """Gets all components from the page that have a 'figure' registered."""
+        return [
+            model_id
+            for model_id in model_manager._get_model_children(model_id=ModelID(str(self.id)))
+            if hasattr(model_manager[model_id], "figure")
+        ]
```

Comment:
This follows our definition here: https://github.com/mckinsey/vizro/pull/114#discussion_r1369160777

but are we sure that this is stable enough, or should this be even more broad and check for registered data?

In any case I think we should explain this "figure" business in a comment

Reply:
Do you think we should call `_has_registered_data` from the `DataManager` instead?
If so, then this is the only place where we make this kind of check (in that case I think we should keep `_has_registered_data` method inside the DataManager then).

Reply:
I am unsure, but wanted to discuss this quickly. So there are two options right?

1. Rely on our taxonomy, and trigger whenever a model has the attribute `figure`?

2. Go via the `DataManager` and check if the component has a registered dataset

On first thought, Option 2 seems more preferable because it is more general, but it may also lead to undesired side effect (e.g. if a registered data set should not have any consequences)

What do you think?

Reply:
For now, I just left the naive approach (`if hasattr(model_manager[model_id], "figure")`) as it works. Once we introduce something like `Figure` protocol then we can change this part of the code as well.